dapp/src/components/buySell/SwapCurrencyPill.js

```diff
@@ -390,6 +391,11 @@ const SwapCurrencyPill = ({
             <CoinSelect
               selected={showOusd ? 'ousd' : selectedCoin}
               onChange={(coin) => {
+                if (coin !== selectedCoin) {
+                  ContractStore.update((s) => {
+                    s.swapEstimations = 'loading'
```

Comment:
These should be triggered as soon as estimations are being ran: https://github.com/OriginProtocol/origin-dollar/blob/master/dapp/src/hooks/useSwapEstimator.js#L182

Any ideas why also the need for it here?

Reply:
Aha I see. That is because there is a 700 ms delay on running the estimations. The reason for that is so that they don't run for each keystroke user does. E.g. you want to swap 2000 stablecoin to OUSD. And they would run for each of those 4 digits. I recommend that you handle this case in this code: 

https://github.com/OriginProtocol/origin-dollar/blob/master/dapp/src/hooks/useSwapEstimator.js#L164

The code in `useSwapEstimator` should recognise that the stablecoin type has been changes or that user has swapped from buying to selling. And in that case not apply 700ms delay. 

The goal is to have all the logic that pertains to the swap delay encapsulated in 1 function in `useSwapEstimator`. It is a lot easier to return to that code in the future and modify it. In the other case if the logic is spread out into various functions it can become very hard to maintain and remember what other parts of the dapp control the logic of delaying estimations.